### PR TITLE
Removed tfio dependency while preserving tensorflow optimization

### PR DESCRIPTION
**Pull request by cverst**

### CODE REVIEW

File: dataloader.py [hunk 1]
```diff
@@ -119,15 +138,15 @@ def load(self, buffer_size: int = None) -> None:
         """
         # if unset, set buffersize to number of samples
         if not buffer_size:
-            buffer_size = self.dataset_input.cardinality().numpy()
+            buffer_size = self._dataset_input.cardinality().numpy()
```
Comment (cverst) [inline dataloader.py]: Missed this one. Should be changed to
`self.n_samples`
I am not sure if this will conflict if
`batch_size > n_samples`

Comment (moreshiny) [reply dataloader.py]: submitted as PR for devc - please merge

**Comment by moreshiny** [comment]
I have suggested some amendments, please merge those into devc and I will merge resulting PR into main @cverst

**Review by moreshiny — CHANGES_REQUESTED** [review]
All good now but please rebase against main @cverst

**Review by moreshiny — APPROVED** [review]

**Pull request merged**